File: baseble/src/main/java/com/fcbox/ble/baseble/callback/IBleCallback.java
package com.fcbox.ble.baseble.callback;


import com.fcbox.ble.baseble.core.BluetoothGattChannel;
import com.fcbox.ble.baseble.exception.BleException;
import com.fcbox.ble.baseble.model.BluetoothLeDevice;


/**
 *
 *  操作数据回调
 *
 */
public interface IBleCallback {
    void onSuccess(byte[] data, BluetoothGattChannel bluetoothGattChannel, BluetoothLeDevice bluetoothLeDevice);

    void onFailure(BleException exception);
}

File: baseble/src/main/java/com/xzy/ble/baseble/callback/IBleCallback.kt
package com.xzy.ble.baseble.callback
import com.xzy.ble.baseble.core.BluetoothGattChannel
import com.xzy.ble.baseble.exception.BleException
import com.xzy.ble.baseble.model.BluetoothLeDevice


/**
 *
 * 操作数据回调
 *
 */
interface IBleCallback {
    fun onSuccess(data: ByteArray, bluetoothGattChannel: BluetoothGattChannel, bluetoothLeDevice: BluetoothLeDevice)

    fun onFailure(exception: BleException)
}
